### PR TITLE
WebGPURenderer: Update attribute only when needed

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
Related issue: #28697

**Description**

This PR addresses an issue where WebGPURenderer was updating all attributes in every render frame, regardless of their state. With this fix, attributes are now updated only if `attribute.needsUpdate` has been called, enhancing performance. /cc @sunag

This PR fixes the handling of interleavedBuffer management.

`InstanceNode` was missing a way to notify the associated buffers if the `instanceMatrix` and `instanceColor` were flagged to be updated, I also added a way to transfer the correct version to the corresponding buffers.

*This contribution is funded by [Utsubo](https://utsubo.com)*

### CODE REVIEW

**Comment by RenaudRohlinger** [comment]
Consequently the example `webgpu_instance_mesh` didn't use `mesh.instanceMatrix.needsUpdate = true;` but was updated regardless since the WebGPURenderer was updating every attributes every render.

I will also need to fix the fact adding `mesh.instanceMatrix.needsUpdate = true;` doesn't seems to fix the current example.

**Comment by RenaudRohlinger** [comment]
Added a fix in `InstanceNode` to notify if the buffers needs to be updated or not.

File: examples/jsm/nodes/accessors/InstanceNode.js [hunk 1]
```diff
@@ -85,6 +93,22 @@ class InstanceNode extends Node {
 
 	}
 
+	update( /*frame*/ ) {
+
+		if ( this.buffer !== null && this.instanceMesh.instanceMatrix.version !== this.buffer.version ) {
```
Comment (sunag) [inline examples/jsm/nodes/accessors/InstanceNode.js:98]: This check necessary? `this.buffer !== null` `update()` should be called after setup.

Comment (RenaudRohlinger) [reply examples/jsm/nodes/accessors/InstanceNode.js]: Indeed! Updated the PR accordingly 👍

**Pull request merged**